### PR TITLE
M3-OBJ-55 Add: Ability to delete buckets

### PR DESCRIPTION
**Pull request by johnwcallahan**
### ~WIP – Not supported by API yet.~  Ready 

Stil to do:

- [x] Write tests
- [x] Test functionality on dev
- [x] "Type name to confirm Delete" UX

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers

To test, use dev services and have Object Storage enabled. Create buckets and delete them.

### CODE REVIEW

File: src/features/ObjectStorage/BucketTableRow.tsx [hunk 1]
```diff
@@ -26,20 +27,31 @@ const styles: StyleRulesCallback<ClassNames> = theme => ({
     display: 'flex',
     flexFlow: 'row nowrap',
     alignItems: 'center'
-  },
-  hostname: { paddingTop: theme.spacing.unit }
+  }
 });
 
 // BucketTableRow has the same props as Linode.Bucket.
 // Aliased for convention's sake.
```
Comment (Jskobos) [inline src/features/ObjectStorage/BucketTableRow.tsx]: No longer really true

File: src/features/ObjectStorage/ListBuckets.tsx [hunk 2]
```diff
@@ -30,10 +43,86 @@ interface Props {
...
+      .then(() => {
+        removeBucketConfirmationDialog.close();
+        setIsLoading(false);
+      })
+      .catch(e => {
+        setIsLoading(false);
+
+        // We're not worried about field errors here, so just grab the text
+        // of the first error in the response.
+        const errorText = pathOr(
```
Comment (Jskobos) [inline src/features/ObjectStorage/ListBuckets.tsx]: `getErrorStringOrDefault` could standardize this.

File: src/features/ObjectStorage/BucketActionMenu.test.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    expect(queryByText('Remove')).toBeInTheDocument();
+  });
+
+  it('executes the onRemove function when the "Remove" option is clicked', () => {
+    const { getByText } = render(
+      wrapWithTheme(<BucketActionMenu {...props} />)
+    );
+
+    fireEvent.click(getByText('Remove'));
+    expect(mockOnRemove).toHaveBeenCalled();
```
Comment (Jskobos) [inline src/features/ObjectStorage/BucketActionMenu.test.tsx]: Would it be meaningful to make this a `toHaveBeenCalledWith()`?

Comment (johnwcallahan) [reply src/features/ObjectStorage/BucketActionMenu.test.tsx]: Great suggestion – thank you.

**Pull request merged**